src/test/java/seedu/letsgethired/testutil/TypicalInternApplications.java

```diff
@@ -120,6 +128,18 @@ public static VersionedInternTracker getTypicalInternTracker() {
     }
 
     public static List<InternApplication> getTypicalInternApplications() {
+        JsonInternTrackerStorage jsonInternTrackerStorage =
+                new JsonInternTrackerStorage(TYPICAL_INTERN_APPLICATIONS_FILE);
+
+
+
+        try {
+            Optional<ReadOnlyInternTracker> internTracker = jsonInternTrackerStorage.readInternTracker();
+            return internTracker.get().getApplicationList();
+        } catch (DataLoadingException dle) {
+            // do nothing
+        }
+
         return new ArrayList<>(Arrays.asList(JANE_STREET, OPTIVER, GOOGLE, META, BYTEDANCE, GRAB, STRIPE));
```

Comment:
Should this be using the newly updated `typicalInternApplicationInternTracker.json` data instead?

Reply:
It doe! Only when it can't read the file, it uses the default data. The reading and returning the file data is in the try block!